### PR TITLE
Activate the sharing of memory between C++ and Python

### PR DESCRIPTION
**Pull request by jcarpent**
This new feature now requires EigenPy >= 2.5.0

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,12 +125,14 @@ ENDIF(BUILD_WITH_CASADI_SUPPORT)
 
 SET(BOOST_REQUIRED_COMPONENTS filesystem serialization system)
 SET(BOOST_BUILD_COMPONENTS unit_test_framework)
-SET(BOOST_OPTIONAL_COMPONENTS "")
+
+ADD_PROJECT_DEPENDENCY(Boost COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
+FIND_PACKAGE(Boost COMPONENTS ${BOOST_BUILD_COMPONENTS})
```
Comment (jmirabel) [inline CMakeLists.txt:130]: Shouldn't there be a *REQUIRED* somewhere here ?
I never know whether it is before or after *COMPONENTS*. I think @nim65s can shade light on this.

Comment (jcarpent) [reply CMakeLists.txt:130]: This is now done. Thanks for the comment.

File: CMakeLists.txt [hunk 2]
```diff
@@ -125,12 +125,15 @@ ENDIF(BUILD_WITH_CASADI_SUPPORT)
 
 SET(BOOST_REQUIRED_COMPONENTS filesystem serialization system)
 SET(BOOST_BUILD_COMPONENTS unit_test_framework)
-SET(BOOST_OPTIONAL_COMPONENTS "")
+
+EXPORT_BOOST_DEFAULT_OPTIONS()
+ADD_PROJECT_DEPENDENCY(Boost REQUIRED COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
+FIND_PACKAGE(Boost COMPONENTS ${BOOST_BUILD_COMPONENTS})
```
Comment (nim65s) [inline CMakeLists.txt:131]: I'd suggest something like
```
IF(BUILD_TESTING)
  FIND_PACKAGE(Boost REQUIRED COMPONENTS ${BOOST_BUILD_COMPONENTS})
ENDIF()
```

File: unittest/python/bindings_force.py [hunk 3]
```diff
@@ -33,12 +33,18 @@ def test_set_linear(self):
         f.linear = lin
         self.assertTrue(np.allclose(f.linear, lin))
 
+        f.linear[1] = 1.
+        self.assertTrue(v.linear[1] == 1.)
```
Comment (wxmerkt) [inline unittest/python/bindings_force.py:37]: Is this meant to be `v` or `f`?

File: CMakeLists.txt (hunk 2 above)
Comment (jcarpent) [reply CMakeLists.txt:131]: I will move it to unit tests.

File: unittest/python/bindings_force.py (hunk 3 above)
Comment (jcarpent) [reply unittest/python/bindings_force.py:37]: You're correct

**Comment by jcarpent** [comment]
As mentioned earlier, this feature requires EigenPy >= 2.5.0, which will be available on robotpkg in the following days. We need to way until then for testing the feature on all the CI.

**Comment by jcarpent** [comment]
Seems things are working well. I still need to expose correctly data fields of Model and Data. I will do it tomorrow ;)

File: unittest/python/bindings_force.py (hunk 3 above)
Comment (jcarpent) [reply unittest/python/bindings_force.py]: Thanks for the feedback.

**Pull request merged**